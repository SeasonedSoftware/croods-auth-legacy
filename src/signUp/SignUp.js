import React, { cloneElement } from 'react'
import New from '../New'

import Created from '../Created'

export default props => {
  const { relativePath = '/auth' } = props

  return <New {...props} name="auth.signUp" path={relativePath} />
}
