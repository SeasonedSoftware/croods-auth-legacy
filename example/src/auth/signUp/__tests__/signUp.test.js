import React from 'react'
import renderer from 'react-test-renderer'
import { New } from 'croods'

import SignUp from '../SignUp'

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - {props.render(props)} - Created
      {props.renderCreated([{ email: 'foo' }])}
    </div>
  ),
}))

jest.mock('@reach/router', () => ({
  Redirect: props => <div {...props}>Redirect</div>,
}))

jest.mock('../Form', () => props => <div {...props}>Form</div>)

jest.mock('../../Created', () => props => <div {...props}>Created</div>)

const props = { setCurrentUser: jest.fn() }
const tree = renderer.create(<SignUp {...props} />)
const newComponent = tree.root.findByType(New)

it('renders correctly', () => {
  expect(tree.toJSON()).toMatchSnapshot()
})

describe('when no currentUser', () => {
  it('redirect', () => {
    const props = {
      user: 'foo',
      currentUser: { email: 'foo' },
      setCurrentUser: jest.fn(),
    }
    const tree = renderer.create(<SignUp {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('when parsing response', () => {
  it('parses response correctly', () => {
    const data = { importantData: { foo: 'bar' } }
    const response = { data, notImportantData: { bar: 'foo' } }

    const parser = newComponent.props.parseResponse

    expect(parser(response)).toMatchObject({ created: data })
  })
})
