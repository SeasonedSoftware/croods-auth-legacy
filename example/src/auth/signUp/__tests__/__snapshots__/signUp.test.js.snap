// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  name="signUp"
  parseResponse={[Function]}
  path="/admin_auth"
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    Form
  </div>
   - Created
  <div
    setCurrentUser={[MockFunction]}
    user={
      Array [
        Object {
          "email": "foo",
        },
      ]
    }
  >
    Created
  </div>
</div>
`;

exports[`when no currentUser redirect 1`] = `
<div
  name="signUp"
  parseResponse={[Function]}
  path="/admin_auth"
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    Form
  </div>
   - Created
  <div
    currentUser={
      Object {
        "email": "foo",
      }
    }
    setCurrentUser={[MockFunction]}
    user={
      Array [
        Object {
          "email": "foo",
        },
      ]
    }
  >
    Created
  </div>
</div>
`;
